packages/rs-dpp/src/state_transition/abstract_state_transition.rs

```diff
@@ -118,11 +120,11 @@ pub trait StateTransitionLike:
             ));
         }
         let data_hash = self.hash(true)?;
-        Ok(signer::verify_hash_signature(
-            &data_hash,
-            self.get_signature(),
-            public_key_hash,
-        )?)
+        signer::verify_hash_signature(&data_hash, self.get_signature(), public_key_hash).or(Err(
```

Comment:
I'd suggest changing all places where you call `or` to `or_else` instead. Also as I understand we don't tell the user why the signature failed, just that it's invalid, right? Maybe losing the message is not always okay, but if that was the original interface, then so be it. Please change `or` to `or_else` anyway :)

Reply:
Fixed.
As of the error message - I tested a couple of different scenarios and generic error was either returning "signature not verified" or something abstract like "bytes array is invalid". I guess we should be fine for most of the cases